diff --git a/jso/impl/src/main/java/org/teavm/jso/impl/JSOPlugin.java b/jso/impl/src/main/java/org/teavm/jso/impl/JSOPlugin.java
index e918744..fd997ea 100644
--- a/jso/impl/src/main/java/org/teavm/jso/impl/JSOPlugin.java
+++ b/jso/impl/src/main/java/org/teavm/jso/impl/JSOPlugin.java
@@ -81,7 +81,7 @@ public class JSOPlugin implements TeaVMPlugin {
             installForJS(jsHost);
         }
 
-        if (wasmGCHost != null) {
+        if (wasmGCHost != null && !Boolean.getBoolean("teavm.wasmgc.nojso")) {
             classTransformer.setClassFilter(n -> !n.startsWith("java."));
             classTransformer.forWasmGC();
             WasmGCJso.install(host, wasmGCHost, repository);
